k8s/charts/openebs/templates/deployment-maya-apiserver.yaml

```diff
@@ -35,18 +35,51 @@ spec:
         # This is supported for maya api server version 0.5.2 onwards
         #- name: OPENEBS_IO_K8S_MASTER
         #  value: "http://172.28.128.3:8080"
+        # OPENEBS_IO_INSTALL_CONFIG_NAME specifies the config map containing the install configuration. 
+        # Currently, the configuration can be used to specifiy the default version for the CAS Templates 
+        - name: OPENEBS_IO_INSTALL_CONFIG_NAME
+          value: {{ template "openebs.fullname" . }}-install-config
+{{- if .Values.ndm.sparse }}
+{{- if .Values.ndm.sparse.enabled }}
+        # OPENEBS_IO_INSTALL_DEFAULT_CSTOR_SPARSE_POOL decides whether default cstor sparse pool should be
+        # configured as a part of openebs installation.
+        # If "true" a default cstor sparse pool will be configured, if "false" it will not be configured.
+        - name: OPENEBS_IO_INSTALL_DEFAULT_CSTOR_SPARSE_POOL
+          value: "{{ .Values.ndm.sparse.enabled }}"
+{{- end }}
+{{- end }}
+        # OPENEBS_NAMESPACE provides the namespace of this deployment as an
+        # environment variable
+        - name: OPENEBS_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        # OPENEBS_SERVICE_ACCOUNT provides the service account of this pod as
+        # environment variable
+        - name: OPENEBS_SERVICE_ACCOUNT
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.serviceAccountName
+        # OPENEBS_MAYA_POD_NAME provides the name of this pod as
+        # environment variable
+        - name: OPENEBS_MAYA_POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
         - name: OPENEBS_IO_JIVA_CONTROLLER_IMAGE
           value: "{{ .Values.jiva.image }}:{{ .Values.jiva.imageTag }}"
         - name: OPENEBS_IO_JIVA_REPLICA_IMAGE
           value: "{{ .Values.jiva.image }}:{{ .Values.jiva.imageTag }}"
+        - name: OPENEBS_IO_CSTOR_TARGET_IMAGE
```

Comment:
Should these image based ENV variables be visible to operators?
IMO maya api server's image controls the images of each micro service.
w.r.t to user/operator's environment, the maya api server's image should control each micro service's image tag.

Reply:
Should we control these bunch of ENV with a debug flag from helm itself. This will hopefully make it clear that this is for debug purposes only.

Reply:
make sense.. as discussed, let us look at how to push this into install-configmap in the coming releases.